packages/ui/src/models/camel/camel-route-resource.ts

```diff
@@ -119,11 +123,17 @@ export class CamelRouteResource implements CamelResource, BeansAwareResource {
     this.serializer = serializer;
   }
 
-  addNewEntity(entityType?: EntityType): string {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  addNewEntity(entityType?: EntityType, entityTemplate?: any): string {
     if (entityType && entityType !== EntityType.Route) {
       const supportedEntity = CamelRouteResource.SUPPORTED_ENTITIES.find(({ type }) => type === entityType);
       if (supportedEntity) {
-        const entity = new supportedEntity.Entity();
+        let entity: BaseVisualCamelEntity;
+        if (entityTemplate) {
+          entity = new supportedEntity.Entity(entityTemplate);
+        } else {
+          entity = new supportedEntity.Entity();
+        }
```

Comment:
Curious question, shouldn't the following serve the same purpose?

```suggestion
          entity = new supportedEntity.Entity(entityTemplate);
```